bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/tag/EnrichedSemanticTagDTO.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.io.rest.core.internal.tag;
+
+import java.util.List;
+
+import org.openhab.core.semantics.SemanticTag;
+
+/**
+ * A DTO representing a {@link SemanticTag}.
+ *
+ * @author Jimmy Tanagra - initial contribution
+ * @author Laurent Garnier - Class renamed and members uid, description and editable added
+ */
+public class EnrichedSemanticTagDTO {
+    String uid;
+    String name;
+    String label;
+    String description;
+    List<String> synonyms;
+    boolean editable;
+    boolean removable;
+
+    public EnrichedSemanticTagDTO(SemanticTag tag, boolean editable, boolean removable) {
+        this.uid = tag.getUID();
+        this.name = tag.getUID().substring(tag.getUID().lastIndexOf("_") + 1);
+        this.label = tag.getLabel();
+        this.description = tag.getDescription();
+        this.synonyms = tag.getSynonyms();
+        this.editable = editable;
+        this.removable = removable;
+    }
```

Comment:
editable means you can update its label, description and synonyms.
removable means you can remove it which means all its child can also be removed. it will avoid removing a tree when inside the tree you have an unmanaged tag.

Reply:
IMO dependencies of non-managed tags on managed tags should not be allowed.

Reply:
And vice versa ?

Reply:
IMO dependencies should only be allowed with the set provided and to those from the `DefaultSemanticTagProvider`, because those are guaranteed to be present, so if provider 1 provides `Location > Bathroom > Mirror` this is possible, because `Location > Bathroom` is provided by the default provider. Provider 2 can add `Location > Bathroom > Shower` but not `Location > Bathroom > Mirror > Property` because `Location > Bathroom > Mirror` is not part of the default tags but from provider 1.

Reply:
Ok, I agree it will be more or less impossible to properly handle all that if you don't add these constraints.
In that case, the "removable" field is no more necessary.
But new checks are needed when creating a new tag with the REST API: the parent must be either a default tag or a managed tag.

Reply:
Removed.
I still need to enhance the control for tag creation.

Reply:
The control in `isNewIdValid` is now added to be able to attach a new tag only to a default tag or managed tag.